--- a/src/glew.c
+++ b/src/glew.c
@@ -23122,17 +23122,6 @@
 #endif /* _WIN32 */
 }
 
-#if defined(_WIN32) && defined(GLEW_BUILD) && defined(__GNUC__)
-/* GCC requires a DLL entry point even without any standard library included. */
-/* Types extracted from windows.h to avoid polluting the rest of the file. */
-int __stdcall DllMainCRTStartup(void* instance, unsigned reason, void* reserved)
-{
-  (void) instance;
-  (void) reason;
-  (void) reserved;
-  return 1;
-}
-#endif
 GLboolean GLEWAPIENTRY glewIsSupported (const char* name)
 {
   const GLubyte* pos = (const GLubyte*)name;
