--- a/loader/loader_windows.c
+++ b/loader/loader_windows.c
@@ -88,7 +88,7 @@
     fpCreateDXGIFactory1 =
         dxgi_module == NULL ? NULL : (PFN_CreateDXGIFactory1)(void *)GetProcAddress(dxgi_module, "CreateDXGIFactory1");

-#if !defined(NDEBUG)
+#if !defined(NDEBUG) && !defined(__MINGW32__)
     _set_error_mode(_OUT_TO_STDERR);
     _CrtSetReportMode(_CRT_ERROR, _CRTDBG_MODE_FILE);
     _CrtSetReportFile(_CRT_ERROR, _CRTDBG_FILE_STDERR);
